### PR TITLE
Code to convert First/rest structure to Collection objects

### PR DESCRIPTION
**Pull request by timbl**
This is needed because CSS tends to serialize lists as first/rest form rather than in Turtle ( parens collections ) form.   This PR does not (yet) add that processing after the parsing of a turtle file

### CODE REVIEW

File: src/collection.ts [hunk 1]
```diff
@@ -112,6 +112,8 @@ export default class Collection<
   }
 
   static toNT (collection) {
+    // return '(' + collection.elements.map(x => x.toNT()).join(' ') + ')'
+    // As list are not in NT and too NT() must be a reversible function, we kludge it for a list
```
Comment (TallTed) [inline src/collection.ts:116]:
```suggestion
    // As lists are not in NT and toNT() must be a reversible function, we kludge it for a list
```

File: src/lists.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import { ContentType, TurtleContentType, N3ContentType, RDFXMLContentType, XHTMLContentType, HTMLContentType, SPARQLUpdateContentType, SPARQLUpdateSingleMatchContentType, JSONLDContentType, NQuadsContentType, NQuadsAltContentType } from './types'
+
+// import { Quad } from './tf-types'
+
+import {BlankNode, NamedNode, Quad, Quad_Graph, Quad_Object, Quad_Predicate, Quad_Subject, Term,} from './tf-types'
+import Namespace from './namespace'
+
+const RDF  = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#')
+
+/* Replace a given node for another node throughout a given document
```
Comment (TallTed) [inline src/lists.ts:30]:
```suggestion
/* Replace a given node with another node throughout a given document
```

**Review by bourgeoa — COMMENTED** [review]
There an issue in npm run build with 
https://stackoverflow.com/questions/55680391/typescript-error-ts2403-subsequent-variable-declarations-must-have-the-same-typ

```
npm uninstall @types/mocha
```
resolves the issue

**Review by bourgeoa — APPROVED** [review]
After removing @types/mocha --> no types issues

else LGTM

**Pull request merged**